src/shaders/symbol_icon.vertex.glsl

```diff
@@ -132,9 +132,11 @@ void main() {
 
     float z = 0.0;
     vec2 offset = rotation_matrix * (a_offset / 32.0 * max(a_min_font_scale, font_scale) + a_pxoffset / 16.0);
+#ifdef TERRAIN
 #ifdef PITCH_WITH_MAP_TERRAIN
```

Comment:
nit: To prevent double define block, how about `#if defined(TERRAIN) && defined(PITCH_WITH_MAP_TERRAIN)`, to also remove the two sequential `#endif` statements.

Reply:
Yes, that would look nicer! The reason why I kept it separate is because our shader cross-compiler script does some replacements on ifdefs with single conditions. But we should eventually update that script to support these compound cases too, I'm going to take a look.

>  In addition to that, are there any useful asserts or validation that we should introduce on the CPU to prevent such invalid use of defines? I did a quick grep of _TERRAIN and this seems to be the only case?

Right, I think we could add a check right before the program creation.